src/ahp/mod.rs

```diff
@@ -215,15 +230,15 @@ impl<F: PrimeField> AHPForR1CS<F> {
         let mut a = LinearCombination::new(
             "a_poly",
             vec![
-                (eta_a * b_denom_at_gamma * c_denom_at_gamma, "a_val"),
-                (eta_b * a_denom_at_gamma * c_denom_at_gamma, "b_val"),
-                (eta_c * b_denom_at_gamma * a_denom_at_gamma, "c_val"),
+                (eta_a * &b_denom_at_gamma * &c_denom_at_gamma, "a_val"),
```

Comment:
All these `&` in the math ops start to get verbose, could we perhaps do the Add/Mul implementations that we talked about, to remove the need for these changes?